File: 523-continuous-subarray-sum/523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mods = {0:-1}
        
        cumsum = 0
        for i in range(len(nums)):
            cumsum += nums[i]
            
            if cumsum % k not in mods:
                mods[cumsum % k] = i
            else:
                if i - mods[cumsum % k] >= 2:
                    return True
        
        return False

File: 523-continuous-subarray-sum/523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> idxMap;
        idxMap[0] = -1;
        int cumSum = 0; 
        for(int i = 0 ; i < nums.size() ; ++i){
            cumSum += nums[i];
            if(idxMap.find(cumSum % k) != idxMap.end()){
                if(i - idxMap[cumSum % k] >= 2){
                    return true;
                }
            }else{
                idxMap[cumSum % k] = i;
            }
        }
        return false;
    }
};